x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/RepeatTests.java

```diff
@@ -0,0 +1,146 @@
+                    new TestCaseSupplier.TypedData(new BytesRef(text), DataTypes.TEXT, "str"),
+                    new TestCaseSupplier.TypedData(number, DataTypes.INTEGER, "number")
+                ),
+                "RepeatEvaluator[str=Attribute[channel=0], number=Attribute[channel=1]]",
+                DataTypes.KEYWORD,
+                equalTo(new BytesRef(text.repeat(number)))
+            );
+        }));
+
+        cases.add(new TestCaseSupplier("Repeat with number zero", List.of(DataTypes.KEYWORD, DataTypes.INTEGER), () -> {
+            String text = randomAlphaOfLength(10);
+            int number = 0;
+            return new TestCaseSupplier.TestCase(
+                List.of(
+                    new TestCaseSupplier.TypedData(new BytesRef(text), DataTypes.KEYWORD, "str"),
+                    new TestCaseSupplier.TypedData(number, DataTypes.INTEGER, "number")
+                ),
+                "RepeatEvaluator[str=Attribute[channel=0], number=Attribute[channel=1]]",
+                DataTypes.KEYWORD,
+                equalTo(new BytesRef(""))
+            );
+        }));
+
+        cases.add(new TestCaseSupplier("Repeat Unicode", List.of(DataTypes.KEYWORD, DataTypes.INTEGER), () -> {
+            String text = randomUnicodeOfLength(10);
+            int number = randomIntBetween(0, 10);
+            return new TestCaseSupplier.TestCase(
+                List.of(
+                    new TestCaseSupplier.TypedData(new BytesRef(text), DataTypes.KEYWORD, "str"),
+                    new TestCaseSupplier.TypedData(number, DataTypes.INTEGER, "number")
+                ),
+                "RepeatEvaluator[str=Attribute[channel=0], number=Attribute[channel=1]]",
+                DataTypes.KEYWORD,
+                equalTo(new BytesRef(text.repeat(number)))
+            );
+        }));
+
+        cases.add(new TestCaseSupplier("Repeat Negative Number", List.of(DataTypes.KEYWORD, DataTypes.INTEGER), () -> {
+            String text = randomAlphaOfLength(10);
+            int number = randomIntBetween(-10, -1);
+            return new TestCaseSupplier.TestCase(
+                List.of(
+                    new TestCaseSupplier.TypedData(new BytesRef(text), DataTypes.KEYWORD, "str"),
+                    new TestCaseSupplier.TypedData(number, DataTypes.INTEGER, "number")
+                ),
+                "RepeatEvaluator[str=Attribute[channel=0], number=Attribute[channel=1]]",
+                DataTypes.KEYWORD,
+                nullValue()
+            ).withWarning("Line -1:-1: evaluation of [] failed, treating result as null. Only first 20 failures recorded.")
+                .withWarning("Line -1:-1: java.lang.IllegalArgumentException: Number parameter cannot be negative, found [" + number + "]");
+        }));
+
+        cases = anyNullIsNull(true, cases);
+        cases = errorsForCasesWithoutExamples(cases);
+        return parameterSuppliersFromTypedData(cases);
+    }
+
+    public void testAlmostTooBig() {
```

Comment:
I left this test as a separate test case because when I put it in a TestCaseSupplier it hits a [circuit breaker](https://github.com/elastic/elasticsearch/blob/6a51c81abe1308a3a02dd6066eb24c94e1bf9f81/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java#L314) here and fails. This seems a bit concerning. Is 1MB perhaps too large a upper bound?

Reply:
We had something similar and @fang-xing-esql spent a while scoping it out. We try to run a test where we copy the input value to a `Block` with anywhere from 1 to 1024 entries in it. So you're MB could end up a GB. And the tests aren't careful about not copying results - I think we duplicate the results. And we limit the tests to 500mb of heap anyway. I'd take these tests and put them in their own file, just so they don't get parameterized for now reason. Just extend `ESTestCase` and leave these two in there. Sorry about the mess.